src/orca/services/nextflowtower/models.py

```diff
@@ -0,0 +1,225 @@
+import json
+from collections.abc import Collection
+from dataclasses import field
+from datetime import datetime
+from typing import Any, Optional
+
+from pydantic.dataclasses import dataclass
+from typing_extensions import Self
+
+
+@dataclass(kw_only=False)
+class User:
+    """Nextflow Tower user."""
+
+    id: int
+    username: str
+    email: str
+
+    @classmethod
+    def from_response(cls, response: dict[str, Any]) -> Self:
+        """Create user from API JSON response.
+
+        Returns:
+            User instance.
+        """
+        return cls(response["id"], response["userName"], response["email"])
+
+
+@dataclass(kw_only=False)
+class Organization:
+    """Nextflow Tower organization."""
+
+    id: int
+    name: str
+
+    @classmethod
+    def from_response(cls, response: dict[str, Any]) -> Self:
+        """Create organization from API JSON response.
+
+        Returns:
+            Organization instance.
+        """
+        return cls(response["orgId"], response["orgName"])
+
+
+@dataclass(kw_only=False)
+class Workspace:
+    """Nextflow Tower workspace."""
+
+    id: int
+    name: str
+    org: Organization
+
+    @property
+    def full_name(self) -> str:
+        """Fully-qualified workspace name (with organization name)."""
+        return f"{self.org.name}/{self.name}".lower()
+
+    @classmethod
+    def from_response(cls, response: dict[str, Any]) -> Self:
+        """Create workspace from API JSON response.
+
+        Returns:
+            Workspace instance.
+        """
+        org = Organization.from_response(response)
+        return cls(response["workspaceId"], response["workspaceName"], org)
+
+
+@dataclass(kw_only=False)
+class LaunchInfo:
+    """Nextflow Tower workflow launch specification."""
+
+    compute_env_id: str
+    pipeline: str
+    work_dir: str
+    revision: Optional[str] = None
+    params: Optional[dict] = None
+    nextflow_config: Optional[str] = None
+    run_name: Optional[str] = None
+    pre_run_script: Optional[str] = None
+    profiles: list[str] = field(default_factory=list)
+    user_secrets: list[str] = field(default_factory=list)
+    workspace_secrets: list[str] = field(default_factory=list)
+    label_ids: list[int] = field(default_factory=list)
+
+    @staticmethod
+    def dedup(items: Collection[str]) -> list[str]:
+        """Deduplicate items in a collection.
+
+        Args:
+            items: Collection of items.
+
+        Returns:
+            Deduplicated collection or None.
+        """
+        return list(set(items))
+
+    @staticmethod
+    def get_current_timestamp():
+        """Generate current timestamp using Tower's UTC format.
+
+        Returns:
+            Current timestamp.
+        """
+        return datetime.now().isoformat()[:-3] + "Z"
+
```

Comment:
YAGNI. I try to avoid implementing methods until I need them. We might be able to use this model once we merge this branch with Brad's `get_workflow()` PR.